int32 label
---
int32 res